core/templates/domain/exploration/StatesObjectFactorySpec.ts

```diff
@@ -336,56 +492,102 @@ describe('States object factory', () => {
       constants.NEW_STATE_TEMPLATE = oldNewStateTemplate;
     });
 
-    it('should create a new state given a state name', () => {
+    it('should create a new state given a state name and set ' +
+      'that state to a terminal state', () => {
       var newStates = ssof.createFromBackendDict(statesDict);
       newStates.addState('new state');
+
       expect(newStates.getState('new state')).toEqual(
-        sof.createFromBackendDict('new state', {
-          classifier_model_id: null,
+        sof.createFromBackendDict('new state', newState2));
+
+      expect(newStates.hasState('new state')).toBe(true);
+      expect(newStates.getStateNames()).toEqual(['first state', 'new state']);
+      expect((Object.keys(newStates.getStateObjects())).length).not.toBe(0);
+
+      newStates.setState('new state',
+        sof.createFromBackendDict('new state', newState));
+
+      expect(newStates.hasState('new state')).toBe(true);
+      expect(newStates.getStateNames()).toEqual(['first state', 'new state']);
+    });
+
+    it('should correctly retrieve the terminal states', () => {
+      var newStates = ssof.createFromBackendDict(statesDict);
+
+      newStates.setState('first state',
+        sof.createFromBackendDict('first state', newState));
+      expect(newStates.getFinalStateNames()).toEqual['new state'];
+    });
+
+    it('should correctly delete a state', () => {
+      var statesWithAudio = ssof.createFromBackendDict(statesWithAudioDict);
+      statesWithAudio.deleteState('first state');
+      expect(statesWithAudio).toEqual(ssof.createFromBackendDict({
+        'second state': secondState
+      }));
+    });
+
+    it('should correctly reset interaction outcomes in the dict ' +
+      'and default outcomes of deleted state', () => {
```

Comment:
I think it's not so clear this one, can you clarify to me what this spec is doing?

Reply:
Done, I clarified the name, basically if there are cyclic default outcomes that point to a deleted or renamed state, this test makes sure that the pointer names are updated correctly.